### ISSUE TITLE
NullReferenceException in CSharpTextPasteHandler.FormatPlainText

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#5642]
```
Text paste handler exception:System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.CSharp.Formatting.CSharpTextPasteHandler.FormatPlainText (System.Int32 insertionOffset, System.String text, System.Byte[] copyData) [0x0004c] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpTextPasteHandler.cs:61 
  at Mono.TextEditor.TextEditorData.PasteText (System.Int32 offset, System.String text, System.Byte[] copyData, System.IDisposable& undoGroup) [0x00008] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:1423 
```

> VS bug [#661699](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/661699)

**Comment by KirillOsenkov** [comment]
Related:

```
Text paste handler exception:System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.CSharp.Formatting.CSharpTextPasteHandler.FormatPlainText (System.Int32 insertionOffset, System.String text, System.Byte[] copyData) [0x0004c] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpTextPasteHandler.cs:61 
  at Mono.TextEditor.TextEditorData.PasteText (System.Int32 offset, System.String text, System.Byte[] copyData, System.IDisposable& undoGroup) [0x00008] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:1423 
Text paste handler exception:System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: charsToReplace
  at Microsoft.VisualStudio.Text.Implementation.BaseBuffer+Edit.Replace (System.Int32 startPosition, System.Int32 charsToReplace, System.String replaceWith) [0x00094] in C:\vs-editor-api\src\Text\Impl\TextModel\BaseBuffer.cs:292 
  at Mono.TextEditor.TextDocument.ApplyTextChanges (System.Collections.Generic.IEnumerable`1[T] changes) [0x0002a] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/Document/TextDocument.cs:429 
  at MonoDevelop.CSharp.Formatting.CSharpTextPasteHandler.FormatPlainText (System.Int32 insertionOffset, System.String text, System.Byte[] copyData) [0x0019a] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpTextPasteHandler.cs:78 
  at Mono.TextEditor.TextEditorData.PasteText (System.Int32 offset, System.String text, System.Byte[] copyData, System.IDisposable& undoGroup) [0x00008] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:1423 

```

**Comment by mkrueger** [comment]
Which branch are you using - line numbers do not match ?

**Comment by KirillOsenkov** [comment]
This is from a customer report log - the build was 7.5 1254.
https://github.com/mono/monodevelop/releases/tag/monodevelop-7.5.0.1254
https://github.com/mono/monodevelop/commit/498923ea36d2c7fe440c4e4b8cfb75bd50bbd748

https://github.com/mono/monodevelop/blob/498923ea36d2c7fe440c4e4b8cfb75bd50bbd748/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpTextPasteHandler.cs#L61

**Comment by mkrueger** [comment]
Ok then it got fixed as side effect with VSTS 634797 fix.

**Comment by KirillOsenkov** [comment]
Fixed in release-7.6 with https://github.com/mono/monodevelop/pull/5198
